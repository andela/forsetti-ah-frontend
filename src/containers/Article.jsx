import React, { Component } from 'react';
import { SingleArticle } from '../components/GetSingleArticle';

class Article extends Component {
  render() {
    return (
      <SingleArticle />
    );
  }
}

export default Article;
