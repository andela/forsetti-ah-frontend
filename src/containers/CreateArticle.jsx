import React from 'react';
import { Container } from 'reactstrap';
import { ArticleFormComponent } from '../components/ArticleForm';
import Footer from '../components/common/Footer';

const CreateArticlePage = () => (
  <Container style={{ minHeight: '600px' }}>
    <ArticleFormComponent />
  </Container>
);

export default CreateArticlePage;
