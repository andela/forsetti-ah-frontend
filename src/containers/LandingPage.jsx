import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Row, Col } from 'reactstrap';
import Skeleton from 'react-skeleton-loader';
import {
  getAritlces,
  getTopRatedArticles,
  openModalAction
} from '../actions';
import {
  Articles,
  SignUpModalComponent,
  HeaderComponent,
  Footer,
  SideDrawerComponent,
  LoginModalComponent,
  ArticleTopRated
} from '../components';

class LandingPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pageCount: 1
    };
  }

  componentDidMount() {
    const {
      getAllArticles,
      topRatedArticles,
      openModal,
      history: {
        action
      }
    } = this.props;
    const mql = window.matchMedia('min-width: 992px').matches;

    const { pageCount } = this.state;
    getAllArticles(pageCount);
    if (!mql) {
      topRatedArticles();
    }

    if (action === 'REPLACE') openModal();
  }

  pageCountHandler = () => {
    const { getAllArticles } = this.props;
    this.setState(prevState => ({
      pageCount: prevState.pageCount + 1
    }), async () => {
      const { pageCount } = this.state;
      await getAllArticles(pageCount);
    });
  }

  render() {
    const {
      articles,
      showSideDrawer: { showSideDrawer },
      modal: { showModal, displayModal },
      topArticles: {
        topRated = []
      }
    } = this.props;
    return (
      <div className='landing-page'>
        <div className='container landing'>
          <div className='row my-5'>
            <div className='col-md-12' />
          </div>
          <div className='row'>
            <div className='col-md-8'>
              {
                articles.isLoading ? (
                  <div className='row'>
                    <div className='col-md-12'>
                      <Skeleton width='100px' height='10px' />
                    </div>
                    <div className='col-md-12'>
                      <Skeleton width='100%' height='100px' count={1} />
                    </div>
                  </div>
                ) : (
                  <Row className=''>
                    <div className='col-md-12'>
                      {
                        articles.articles && (
                        <Articles
                          articles={articles.articles}
                          loadMore={this.pageCountHandler}
                        />
                        )
                      }
                    </div>
                  </Row>
                )
              }
            </div>
            <div className='col-md-4 mt-4 top'>
              <h4
                className='border border-top-0 border-left-0 border-right-0 border-dark pb-2'
              >
      Top Articles
              </h4>
              <div className='col-12 mt-2' style={{ height: '32rem', overflow: 'scroll' }}>
                {
        topRated.length !== 0 ? (
          topRated.map(top => (
            <ArticleTopRated
              key={top.id}
              title={top.title}
              description={top.description}
              readingTime={top.readingTime}
              image={top.author ? top.author.image : ''}
              firstname={top.author ? top.author.firstname : ''}
              username={top.author ? top.author.username : ''}
              slug={top.slug}
            />
          ))
        ) : (<p>No Top Articles</p>)

      }
              </div>
            </div>
          </div>
        </div>
        <SignUpModalComponent isOpen={displayModal} />
        <Row className='articles'>
          <Col sm={{ size: 6 }} />
          <Col sm={{ size: 4 }} />
        </Row>
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  const {
    showSideDrawer,
    modal,
    articles,
    topArticles
  } = state;
  return {
    articles,
    showSideDrawer,
    modal,
    topArticles
  };
};
const mapDispatchToProps = {
  getAllArticles: getAritlces,
  topRatedArticles: getTopRatedArticles,
  openModal: openModalAction
};

const LandingPageComponent = connect(mapStateToProps, mapDispatchToProps)(LandingPage);

export {
  LandingPage,
  LandingPageComponent
};
