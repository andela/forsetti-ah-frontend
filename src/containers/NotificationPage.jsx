import React, { Component } from 'react';
import { connect } from 'react-redux';
import Skeleton from 'react-skeleton-loader';
import { Link } from 'react-router-dom';
import { getUserNotifications } from '../actions';
import NotificationCard from '../components/common/noficationCard';
/**
 * Notification Component
 */
export class Notification extends Component {
  async componentDidMount() {
    const { getNotifications } = this.props;
    await getNotifications();
  }

  render() {
    const {
      notifications: {
        notifications = [],
        isLoading = false
      }
    } = this.props;
    let data;
    if (isLoading) {
      data = (
        <div className='h-75 col-12 col-md-10'>
          <Skeleton
            count={4}
            width='100%'
            height='6rem'
            widthRandomness={0}
            borderRadius={0}
          />
        </div>
      );
    } else if (isLoading === false && notifications.length === 0) {
      data = (
        <h2>You have no notications</h2>
      );
    } else {
      data = (
        notifications && notifications.map(({ id, ...notices }) => (
          <NotificationCard
            key={id}
            {...notices}
          />
        ))
      );
    }
    return (
      <div className='container notification-page'>
        <div className='row py-4 mx-1 mx-md-0'>
          <h2 className='notification-header'>Notifications</h2>
        </div>
        <div className='row mx-1 mx-md-0'>
          {data}
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  notifications: state.notifications,
});

const mapDispatchToProps = {
  getNotifications: getUserNotifications
};

export default connect(mapStateToProps, mapDispatchToProps)(Notification);
