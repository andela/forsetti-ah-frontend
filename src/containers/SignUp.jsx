import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { Modal, ModalBody, ModalHeader, InputGroupAddon, InputGroup, Input, InputGroupText } from 'reactstrap';
import signup from '../assets/images/signup.svg';
import ButtonComponent from '../components/common/Button';
import { signUpUser } from '../actions';
import FormErrorText from '../components/common/FormErrorText';
import { checkInputValues, clearErrorMessages } from '../helpers/signupFormValidator';

export class SignUp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: true,
      firstname: '',
      firstnameError: '',
      lastname: '',
      lastnameError: '',
      email: '',
      emailError: '',
      password: '',
      passwordError: '',
      username: '',
      usernameError: '',
    };
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const {
      firstname, lastname, email, password, username
    } = this.state;
    const data = {
      firstname,
      lastname,
      email,
      password,
      username
    };
    const errorMessages = checkInputValues(data);
    const errorFields = Object.keys(errorMessages);
    this.setState(clearErrorMessages());

    if (errorFields.length > 0) {
      this.setState(errorMessages);
    } else {
      const emptyErrorMessages = clearErrorMessages();
      this.setState(emptyErrorMessages);

      const { dispatch } = this.props;
      dispatch(signUpUser(data));
    }
  }

  toggle = () => {
    this.setState(prevState => ({
      modal: !prevState.modal
    }));
  }

  render() {
    const {
      modal,
      firstname,
      lastname,
      email,
      password,
      username,
      firstnameError,
      lastnameError,
      emailError,
      passwordError,
      usernameError
    } = this.state;

    const { redirect } = this.props;
    if (redirect) return <Redirect to='/' />;

    return (
      <div id='signup'>
        <Modal isOpen={modal} toggle={this.toggle} className='modal-dialog'>
          <ModalHeader
            toggle={this.toggle}
            style={{
              borderBottom: 0,
            }}
          />
          <ModalBody className='modal-body'>
            <div className='d-block d-md-flex d-lg-flex'>
              <form className='w-100' onSubmit={this.handleSubmit}>

                <InputGroup className='signup-form-group'>
                  <InputGroupAddon addonType='prepend'>
                    <InputGroupText
                      className='signup-form-label'
                    >
                      FIRSTNAME
                    </InputGroupText>
                  </InputGroupAddon>
                  <Input
                    placeholder='firstname'
                    type='text'
                    className='signup-form-input'
                    onChange={this.handleChange}
                    name='firstname'
                    value={firstname}
                  />
                </InputGroup>
                <FormErrorText message={firstnameError} className='form-error-text' />

                <InputGroup className='signup-form-group'>
                  <InputGroupAddon addonType='prepend'>
                    <InputGroupText
                      className='signup-form-label'
                    >
                      LASTNAME
                    </InputGroupText>
                  </InputGroupAddon>
                  <Input
                    placeholder='lastname'
                    type='text'
                    className='signup-form-input'
                    onChange={this.handleChange}
                    name='lastname'
                    value={lastname}
                  />
                </InputGroup>
                <FormErrorText message={lastnameError} className='form-error-text' />

                <InputGroup className='signup-form-group'>
                  <InputGroupAddon addonType='prepend'>
                    <InputGroupText
                      className='signup-form-label'
                    >
                      EMAIL
                    </InputGroupText>
                  </InputGroupAddon>
                  <Input
                    placeholder='email'
                    type='email'
                    className='signup-form-input'
                    onChange={this.handleChange}
                    name='email'
                    value={email}
                  />
                </InputGroup>
                <FormErrorText message={emailError} className='form-error-text' />

                <InputGroup className='signup-form-group'>
                  <InputGroupAddon addonType='prepend'>
                    <InputGroupText
                      className='signup-form-label'
                    >
                      PASSWORD
                    </InputGroupText>
                  </InputGroupAddon>
                  <Input
                    placeholder='password'
                    type='password'
                    className='signup-form-input'
                    onChange={this.handleChange}
                    name='password'
                    value={password}
                  />
                </InputGroup>
                <FormErrorText message={passwordError} className='form-error-text' />

                <InputGroup className='signup-form-group'>
                  <InputGroupAddon addonType='prepend'>
                    <InputGroupText
                      className='signup-form-label'
                    >
                      USERNAME
                    </InputGroupText>
                  </InputGroupAddon>
                  <Input
                    placeholder='username'
                    type='text'
                    className='signup-form-input'
                    onChange={this.handleChange}
                    name='username'
                    value={username}
                  />
                </InputGroup>
                <FormErrorText message={usernameError} className='form-error-text' />

                <div>
                  <ButtonComponent
                    className='signup-form-button'
                  >
                    JOIN NOW
                  </ButtonComponent>
                </div>
              </form>
              <div className='w-50 d-none d-md-block'>
                <img src={signup} alt='signup-icon' />
              </div>
            </div>

          </ModalBody>
        </Modal>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  redirect: state.auth.redirect,
  isloading: state.auth.isloading,
  submit: state.auth.submit,
  firstname: state.auth.firstname,
  lastname: state.auth.lastname,
  password: state.auth.password,
  email: state.auth.email,
  username: state.auth.username,
  errors: state.auth.errors
});

export default connect(mapStateToProps)(SignUp);
