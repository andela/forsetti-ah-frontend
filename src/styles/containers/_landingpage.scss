.landing-page {
  .col-lg-12 {
    padding-right: 0;
  }
}

.landing {
  min-height: 100vh;
}
.top {
  display: none;
}

@media screen and (min-width: 992px) {
  .top {
    display: block;
  }
}