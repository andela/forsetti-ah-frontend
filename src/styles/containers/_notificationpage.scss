.notification-page {
  min-height: 80vh;
}
@media screen and (max-width: 575px) {
  .notification-icon {
    display: none;
  }
}