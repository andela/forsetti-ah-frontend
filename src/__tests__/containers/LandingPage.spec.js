import React from 'react';
import { shallow } from 'enzyme';
import { articles as articlesMock } from '../../testUtils/testsMockData/articles.mock-data';
import { LandingPage } from '../../containers/LandingPage';

describe('<LandingPage />', () => {
  const props = {
    showSideDrawer: {
      showSideDrawer: false,
    },
    modal: {
      showModal: false,
    },
    articles: {
      isLoading: false,
      aritcles: articlesMock
    },
    getAllArticles: jest.fn(),
    history: {
      action: 'POP'
    },
    openModal: jest.fn()
  };

  it('renders correctly', () => {
    const LandingPageComponent = shallow(<LandingPage {...props} />);
    expect(LandingPageComponent).toMatchSnapshot();
  });

  it('opens login modal on redirect', () => {
    const props = {
      showSideDrawer: {
        showSideDrawer: false,
      },
      modal: {
        showModal: false,
      },
      articles: {
        isLoading: false,
        aritcles: articlesMock
      },
      getAllArticles: jest.fn(),
      history: {
        action: 'REPLACE'
      },
      openModal: jest.fn()
    };
    const LandingPageComponent = shallow(<LandingPage {...props} />);
    expect(props.openModal).toHaveBeenCalled();
  })
});
