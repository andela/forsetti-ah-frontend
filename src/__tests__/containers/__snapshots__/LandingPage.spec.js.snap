// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders correctly 1`] = `
<div
  className="landing-page"
>
  <div
    className="backdrop"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="sidedrawer"
    style={
      Object {
        "opacity": "0",
        "visibility": "hidden",
      }
    }
  >
    <div
      className="sidedrawer-nav"
    >
      <div
        className="sidedrawer-nav-close-icon"
      >
        <button
          aria-label={null}
          className="sidedrawer-nav-close-button btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          <i
            className="far fa-window-close"
          />
        </button>
      </div>
      <div
        className="sidedrawer-nav-links"
      >
        <ul
          className="sidedrawer-nav-links-list"
        >
          <li>
            Tech
          </li>
          <li>
            Philosophy
          </li>
          <li>
            Life
          </li>
          <li>
            Politics
          </li>
          <li>
            Nature
          </li>
          <li>
            Music
          </li>
        </ul>
        <ul
          className="sidedrawer-nav-menu-list"
        >
          <li>
            <button
              aria-label={null}
              className="navbar-button-item btn btn-primary"
              onClick={[Function]}
            >
              Sign in
            </button>
          </li>
          <li>
            <button
              aria-label={null}
              className="navbar-button-item btn btn-primary"
              onClick={[Function]}
            >
              Sign up
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <nav
        className="header d-flex justify-content-between"
      >
        <div
          className="header-logo"
        >
          <h1
            className="header-logo-text"
          >
            Authors Haven
          </h1>
        </div>
        <div
          className="header-links"
        >
          <ul
            className="header-links-list"
          >
            <li>
              Tech
            </li>
            <li>
              Philosophy
            </li>
            <li>
              Life
            </li>
            <li>
              Politics
            </li>
            <li>
              Nature
            </li>
            <li>
              Music
            </li>
          </ul>
        </div>
        <div
          className="header-nav"
        >
          <ul
            className="header-nav-menu"
          >
            <li>
              <button
                aria-label={null}
                className="navbar-button-item btn btn-primary"
                onClick={[Function]}
              >
                Sign in
              </button>
            </li>
            <li>
              <button
                aria-label={null}
                className="navbar-button-item btn btn-primary"
                onClick={[Function]}
              >
                Sign up
              </button>
            </li>
          </ul>
          <button
            aria-label={null}
            className="header-nav-hamburger btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-bars"
            />
          </button>
        </div>
      </nav>
    </div>
  </div>
  <div
    className="articles row"
  >
    <div
      className="col-sm-6"
    />
    <div
      className="col-sm-4"
    />
  </div>
  <div
    className="footer row"
  >
    <div
      className="col"
    >
      <footer
        className="footer"
      >
        <div
          className="footer-upper-section row"
        >
          <div
            className="footer-logo col-sm-12 col-lg-4"
          >
            <h6
              className="footer-logo-text"
            >
              Authors Haven
            </h6>
          </div>
          <div
            className="footer-links col-sm-4"
          >
            <div
              className="footer-links-box"
            >
              <ul
                className="footer-links-box-list"
              >
                <li>
                  Tech
                </li>
                <li>
                  Philosophy
                </li>
                <li>
                  Life
                </li>
                <li>
                  Politics
                </li>
              </ul>
              <ul
                className="footer-links-box-list"
              >
                <li>
                  Politics
                </li>
                <li>
                  Culture
                </li>
                <li>
                  Nature
                </li>
                <li>
                  Music
                </li>
              </ul>
              <ul
                className="footer-links-box-list"
              >
                <li>
                  Finance
                </li>
                <li>
                  Adventure
                </li>
                <li>
                  Motivational
                </li>
                <li>
                  Spiritual
                </li>
              </ul>
            </div>
          </div>
          <div
            className="footer-top-logo col-sm-4"
          >
            <i
              className="fas fa-arrow-up footer-top-logo-icon"
            />
          </div>
        </div>
        <div
          className="footer-lower-section row"
        >
          <div
            className="col"
          >
            <p
              className="footer-lower-section-text"
            >
               Â© 2019
            </p>
          </div>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
